src/nix/repl.cc

```diff
@@ -895,6 +895,22 @@ struct CmdRepl : InstallablesCommand
     CmdRepl(){
         evalSettings.pureEval = false;
     }
+    void prepare()
+    {
+        if (!settings.isExperimentalFeatureEnabled(Xp::Flakes) && !(file)) {
```

Comment:
Shouldn’t we have a separate XP feature for that? I’m imagining that this could become the default way before flakes become non-xp

Reply:
perhaps with "nix-command" instead?

Reply:
I’d rather have it use its own `repl-flake` experimental feature. It’s cheap and that will leave us with more freedom to flip the switch soon-ish